shared/chat/hud/index.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+import React from 'react'
+import {connect} from 'react-redux'
+import {compose, withState, lifecycle, withPropsOnChange, branch, renderNothing} from 'recompose'
+import {Avatar, Box, ClickableBox, List, Text} from '../../common-adapters/index'
+import {globalColors, globalMargins, globalStyles} from '../../styles'
+
+import type {TypedState} from '../../constants/reducer'
+
+type Props<D: {key: string, selected: boolean}> = {
+  rowRenderer: (i: number, d: D) => React$Element<*>,
+  data: Array<D>,
+  style: Object,
+}
+
+type MentionDatum = {
+  avatar: React$Element<*>,
+  username: string,
+  fullName: string,
+  selected: boolean,
+  onClick: () => void,
+  onHover: () => void,
+}
+
+const MentionRowRenderer = ({avatar, username, fullName, selected, onClick, onHover}: MentionDatum) => (
+  <ClickableBox
+    style={{
+      ...globalStyles.flexBoxRow,
+      height: 24,
+      alignItems: 'center',
+      paddingLeft: globalMargins.small,
+      paddingRight: globalMargins.small,
+      backgroundColor: selected ? globalColors.blue4 : undefined,
+    }}
+    onClick={onClick}
+    onMouseOver={onHover}
+  >
+    {avatar}
+    <Text type="Body" style={{marginLeft: globalMargins.tiny}}>{username}</Text>
+    <Text type="Body" style={{marginLeft: globalMargins.tiny}}>{fullName}</Text>
+  </ClickableBox>
+)
+
+const Hud = ({style, data, rowRenderer}: Props<*>) => (
+  <Box style={{...hudStyle, ...style}}>
+    <List items={data} renderItem={rowRenderer} />
```

Comment:
Would using ReactList directly here make it easier to access the [scrollAround(i)](https://github.com/orgsync/react-list#scrollaroundindex) fcn to auto-scroll when the user moves into the overflow?

Reply:
List is meant to be a simple cross platform interface. Maybe the better thing would be to add scrollAround functionality to `List` and use `scrollAround` under the hood (and implement something similar for mobile.

But it's also not obvious why you'd want that for this?

Reply:
Ah it's so you can see the visible item if you get to it from kb shortcuts.

Reply:
Also as it is when the list of autocompletes is longer than the hud container (i.e. typing `@m` in keybasefriends) and you arrowDown past the container, you'd still want to see the highlighted profile. When I tested the behavior was that it'd just select past the container into items that aren't visible. But yeah I agree that putting `scrollAround` in List would make sense